src/login.rs

```diff
@@ -54,14 +58,54 @@ pub async fn perform_login(config: LoginCLIConfig, webdriver: &WebDriver) -> Res
     debug!("found login button");
 
     login_button.click().await.context("click login button")?;
-    debug!("clicked login button, waiting for login");
+    debug!("clicked login button, checking for one-time-code or login");
+    
+    // Sloppily re-defined onetimecode var here because of "ownership" issues in the loop (Rust noob)
+    let onetimecode = config.onetimecode;
+    // loggedin tracks if timeout loop exited due to successful login or due to 2FA detection
+    let mut loggedin = false;
+    
+    // Check if login successful or if 2FA required (Cancel button appears)    
+    tokio::time::timeout(Duration::from_secs(20), async {
+        loop {
+            if has_new_email_button(webdriver)
+                .await
+                .context("search new-email button")?
+            {
+            	loggedin = true;
+                return Ok::<_, anyhow::Error>(()); // Login successful
+            }
+            else if has_cancel_button(webdriver)
+                .await
+                .context("search cancel button")?
+            {
+            	// Cancel button present means 2FA needed
+            	enter_onetimecode(&onetimecode, webdriver)
+            	    .await
+            	    .context("entering one-time-code and finishing login")?;
+            	return Ok::<_, anyhow::Error>(()); // 2FA entered, logging in...
+            }
 
+            tokio::time::sleep(Duration::from_secs(1)).await;
+        }
+    })
+    .await
+    .context("wait for one-time-code or login")??;
```

Comment:
```suggestion
    // Check if login successful or if 2FA required (Cancel button appears)    
    let loggedin = tokio::time::timeout(Duration::from_secs(20), async {
        loop {
            if has_new_email_button(webdriver)
                .await
                .context("search new-email button")?
            {
                // Login successful
                return Ok::<_, anyhow::Error>(true);
            }
            else if has_cancel_button(webdriver)
                .await
                .context("search cancel button")?
            {
            	// Cancel button present means 2FA needed
            	enter_onetimecode(&onetimecode, webdriver)
            	    .await
            	    .context("entering one-time-code and finishing login")?;
            	// 2FA entered, logging in...
            	return Ok::<_, anyhow::Error>(false);
            }

            tokio::time::sleep(Duration::from_secs(1)).await;
        }
    })
    .await
    .context("wait for one-time-code or login")??;
```

This is slightly more "Rusty".